### PR TITLE
minor style changes

### PR DESCRIPTION
**Pull request by CumpsD**
I did a bunch of minor (OCD) style changes.
 * newline at the end, and making the classes inside FsCodec.fs look the same ;)
 * making usage of comma's in Codec.fs consistent with FsCodec.fs, easier on the eyes
 * making the type hints consistent wrt spacing
 * making tuple definitions consistent wrt spacing

Some additional remarks:
* A lot of type hints the compiler can actually figure out itself, could reduce a lot of mental noise. Didn't want to touch that yet since it's a bigger impact, maybe you prefer seeing the type explicit
* Being more liberal with newlines makes the code more _airy_ and easier to read, ask [Kevlin Henney](https://www.youtube.com/watch?v=ZsHMHukIlJY) ;)

PS: Feel free to squash these all into one, or let me know if you want me to do it

### CODE REVIEW

**Comment by bartelink** [comment]
Nice work - I actually like the missing EOLs at EOF (you'll find same in Equinox and Propulsion), but you've done a very nice job so all good (🤔 maybe a .editorconfig can/should pin it?)
(I'll squash merge when you're ready - I do that for most things)

**Comment by CumpsD** [comment]
@bartelink it depends, how far do you want me to go? Have a look at this nice talk: https://www.youtube.com/watch?v=ZsHMHukIlJY

I would personally add a lot of newlines in there to make things easier to explore, but I dont want to do that before knowing you are comfortable with it.

I'll have a look at the editorconfig

**Comment by bartelink** [comment]
>A lot of type hints the compiler can actually figure out itself, could reduce a lot of mental noise. Didn't want to touch that yet since it's a bigger impact, maybe you prefer seeing the type explicit

That's an intentional style thing - being friendly to the reader; and Explicit is Better than Implicit at times. (The other thing is that there's nothing like apidiff to trap subtle contract changes in here atm so its pretty important.)

>Being more liberal with newlines makes the code more airy and easier to read, ask Kevlin Henney ;)

Yes, aware of the talk and thinking and have no great arguments against. I always love when people lay things out properly (shoutout to Eirik and Kostas!) but have an OCD affliction about doing minimal LOC. For some kinds of code, making stuff that conceptually should not be split fit one one screen is something I'll pay a higher price for than most more reasonable folk!

**Comment by CumpsD** [comment]
@bartelink in that case, this PR is done and ready for review ;)

**Comment by bartelink** [comment]
>I would personally add a lot of newlines in there to make things easier to explore, but I dont want to do that before knowing you are comfortable with it.

In general, I'd say most of my formatting is pretty deliberate - I don't take as Seemann / Henney a stance as some - try me, but please lean on the conservative side...

**Comment by CumpsD** [comment]
@bartelink I'm done with the PR 👍

File: src/FsCodec.NewtonsoftJson/Serdes.fs [hunk 1]
```diff
@@ -31,6 +31,6 @@ type Serdes private () =
         (   /// Json string to deserialize.
             json : string,
             /// Settings to use (defaults to Settings.Create() profile)
-            [<Optional; DefaultParameterValue null>]?settings : JsonSerializerSettings) : 'T =
+            [<Optional; DefaultParameterValue null>]? settings : JsonSerializerSettings) : 'T =
```
Comment (bartelink) [inline src/FsCodec.NewtonsoftJson/Serdes.fs]: omg an inconsistent fix!

File: src/FsCodec.NewtonsoftJson/OptionConverter.fs [hunk 2]
```diff
@@ -8,17 +8,18 @@ open System
 type OptionConverter() =
     inherit JsonConverter()
 
-    override __.CanConvert(t) = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<option<_>>
+    override __.CanConvert(t : Type) = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<option<_>>
 
-    override __.WriteJson(writer, value, serializer) =
+    override __.WriteJson(writer : JsonWriter, value : obj, jsonSerializer : JsonSerializer) =
```
Comment (bartelink) [inline src/FsCodec.NewtonsoftJson/OptionConverter.fs]: think `serializer` was a better name, aligning with `writer` etc. elsewhere

File: src/FsCodec.NewtonsoftJson/Codec.fs [hunk 3]
```diff
@@ -69,33 +71,36 @@ type Codec private () =
...
                 let (c, meta : 'Meta option, correlationId, causationId, timestamp : DateTimeOffset option) = down (context, event)
                 let enc = dataCodec.Encode c
                 let metaUtf8 = meta |> Option.map bytesEncoder.Encode<'Meta>
-                FsCodec.Core.EventData.Create(enc.CaseName, enc.Payload, defaultArg metaUtf8 null, correlationId, causationId, ?timestamp=timestamp)
+                FsCodec.Core.EventData.Create(enc.CaseName, enc.Payload, defaultArg metaUtf8 null, correlationId, causationId, ?timestamp = timestamp)
+
             member __.TryDecode encoded =
                 let cOption = dataCodec.TryDecode { CaseName = encoded.EventType; Payload = encoded.Data }
-                match cOption with None -> None | Some contract -> let union = up (encoded,contract) in Some union }
+                match cOption with None -> None | Some contract -> let union = up (encoded, contract) in Some union }
```
Comment (bartelink) [inline src/FsCodec.NewtonsoftJson/Codec.fs]: maybe inline cOption , and break the match cases into a oneliner for None and 3 lines for the Some?

File: src/FsCodec.NewtonsoftJson/TypeSafeEnumConverter.fs [hunk 4]
```diff
@@ -7,26 +7,28 @@ open System
...
-    let tryParseT (t: Type) (str: string) =
+    let tryParseT (t : Type) (str : string) =
         let union = Union.getUnion t
         union.cases
         |> Array.tryFindIndex (fun case -> case.Name = str)
         |> Option.map (fun tag -> (union.caseConstructor.[tag] [||]))
-    let tryParse<'T> (str: string) = tryParseT typeof<'T> str |> Option.map (fun e -> e :?> 'T)
 
-    let parseT (t: Type) (str: string) =
+    let tryParse<'T> (str : string) = tryParseT typeof<'T> str |> Option.map (fun e -> e :?> 'T)
```
Comment (bartelink) [inline src/FsCodec.NewtonsoftJson/TypeSafeEnumConverter.fs]: its fine, but I was trying to keep the two 'overloads' of parse and tryParse were being aligned visually the way it was?

File: src/FsCodec.NewtonsoftJson/OptionConverter.fs (hunk 2 above)
Comment (CumpsD) [reply src/FsCodec.NewtonsoftJson/OptionConverter.fs]: Everything else was named `jsonSerializer`, that's why I took it :) but I agree, `serializer` is better, will change

**Comment by bartelink** [comment]
Thanks; it'll be great for other readers not to have to endure idiosyncratic formatting, no matter how much I once managed to convince myself they were a good idea!

**Pull request merged**